dc-cudami-admin/dc-cudami-admin-webapp/src/main/java/de/digitalcollections/cudami/admin/controller/identifiable/entity/parts/WebpagesController.java

```diff
@@ -138,6 +142,22 @@ public String view(@PathVariable UUID uuid, Model model) {
     List<FileResource> relatedFileResources = service.getRelatedFileResources(webpage);
     model.addAttribute("relatedFileResources", relatedFileResources);
 
+    List<Node> breadcrumbs = new ArrayList<>();
+    addParentNodeToBreadcrumb(webpage, breadcrumbs);
+    Collections.reverse(breadcrumbs);
+    model.addAttribute("breadcrumbs", breadcrumbs);
+
+    Website website = service.getWebsite(uuid);
+    model.addAttribute("website", website);
+
     return "webpages/view";
   }
+
+  private void addParentNodeToBreadcrumb(Node currentWebpage, List<Node> breadcrumbs) {
+    Node parent = service.getParent(currentWebpage);
+    if (parent != null && parent.getUuid() != null) {
+      breadcrumbs.add(parent);
+      addParentNodeToBreadcrumb(parent, breadcrumbs);
+    }
```

Comment:
Auch wieder ne Datenbankquery im Loop, geht das nicht im SQL?

Reply:
https://learnsql.com/blog/do-it-in-sql-recursive-tree-traversal/

Reply:
danke für den tipp. würde ich in ein verbesserungsticket auslagern wollen

Reply:
OK, wäre aber schon gut wenn das zeitnah passieren würde, das ist ja auch ein Feature das wir in Endanwendungen verwenden müssen, und der Performance/Latenz-Impact von dem Pattern hier ist schon relativ beträchtlich fürchte ich.

Reply:
ist nun im backlog: https://jira.digitale-sammlungen.de/browse/MR-960